src/main/java/com/amazon/ion/system/IonTextWriterBuilder_1_1.java

```diff
@@ -0,0 +1,367 @@
+     * returning a new mutable builder if this is immutable.
+     *
+     * @param charset may be null, denoting the default of UTF-8.
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     *
+     * @see #getCharset()
+     * @see #setCharset(Charset)
+     */
+    IonTextWriterBuilder_1_1 withCharset(Charset charset);
+
+    /**
+     * Declares the output encoding to be {@code US-ASCII}.
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     */
+    IonTextWriterBuilder_1_1 withCharsetAscii();
+
+    /**
+     * Declares that this builder should minimize system-level output
+     * (Ion version markers and local symbol tables).
+     * <p>
+     * This is equivalent to:
+     * <ul>
+     *   <li>{@link #setIvmMinimizing(IonWriterBuilder.IvmMinimizing)
+     *   setIvmMinimizing}{@code (}{@link IonWriterBuilder.IvmMinimizing#DISTANT DISTANT}{@code )}
+     *   <li>{@link #setLstMinimizing(IonTextWriterBuilder.LstMinimizing)
+     *   setLstMinimizing}{@code (}{@link IonTextWriterBuilder.LstMinimizing#EVERYTHING EVERYTHING}{@code )}
+     * </ul>
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     *
+     */
+    IonTextWriterBuilder_1_1 withMinimalSystemData();
+
+
+    /**
+     * Declares that this builder should use basic pretty-printing.
+     * Does not alter the handling of system data.
+     * Calling this method alters several other configuration properties,
+     * so code should call it first, then make any necessary overrides.
+     * <p>
+     * The specifics of this configuration may change between releases of this
+     * library, so automated processes should not depend on the exact output
+     * formatting.
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     */
+    IonTextWriterBuilder_1_1 withPrettyPrinting();
+
+    /**
+     * Declares that this builder should downgrade the writers' output to
+     * JSON compatibility. This format cannot round-trip back to Ion with full
+     * fidelity.
+     * <p>
+     * The specific conversions are as follows:
+     * <ul>
+     *   <li>System data is suppressed per {@link #withMinimalSystemData()}.
+     *   <li>All annotations are suppressed.
+     *   <li>Nulls of any type are printed as JSON {@code null}.
+     *   <li>Blobs are printed as strings, containing Base64.
+     *   <li>Clobs are printed as strings, containing only Unicode code points
+     *       U+00 through U+FF.
+     *   <li>Sexps are printed as lists.
+     *   <li>Symbols are printed as strings.
+     *   <li>Timestamps are printed as strings, using Ion timestamp format.
+     * </ul>
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     */
+    IonTextWriterBuilder_1_1 withJsonDowngrade();
```

Comment:
It sounds like this might also imply write-time evaluation of E-Expressions so that they won't appear in the serialized data. We should make that explicit.

Reply:
This JavaDoc points to `withMinimalSystemData()`. I added a note there about expansion of macros.